axelrod/strategy_transformers.py

```diff
@@ -228,11 +297,21 @@ def dual_wrapper(player, opponent, proposed_action):
     -------
     action: an axelrod.Action, C or D
     """
+
     if not player.history:
-        player.original_player = player.original_class(**player.init_kwargs)
+        player.use_history = []
```

Comment:
i'll comment the code where state-change is recorded, and the possible hidden issue (that was also true with the original)

here's the test i just tried. `assert_dual_wrapper_correct` is lifted from code in `test_strategy_transformers.py`.  (it was a separate method as I used it in a few other tests (which can be mostly be removed)).

```
    def test_all_strategies_with_dual(self):
        for s in axl.strategies:
            self.assert_dual_wrapper_correct(s)

    def assert_dual_wrapper_correct(self, player_class):
        p1 = player_class()
        p2 = st.DualTransformer()(player_class)()
        p3 = axl.CyclerCCD()  # Cycles 'CCD'

        axl.seed(0)
        for _ in range(10):
            p1.play(p3)

        p3.reset()

        axl.seed(0)
        for _ in range(10):
            p2.play(p3)

        self.assertEqual(p1.history, [x.flip() for x in p2.history])
```

it failed on  `fails = [axl.EvolvedANN , axl.EvolvedANN5, axl.EvolvedANNNoise05, axl.Golden, axl.e]`

after i added a function to switch player.cooperations and player.defections (and then switch them back) they all passed.  

Does this work, testing-wise?